homeassistant/components/onewire/sensor.py

```diff
@@ -224,6 +242,16 @@ def unit_of_measurement(self):
         """Return the unit the value is expressed in."""
         return self._unit_of_measurement
 
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the sensor."""
+
+        return {
+            "device_file" : self._device_file,
+            "sensor_type": self._sensor_type,
+            "raw_value": self._value_raw
```

Comment:
Please describe how these attributes will be used by the user. Sensor type and raw value seem of limited value. Sensor type is already part of the sensor name.

Reply:
raw_value is of use as the sensor state gives back value rounded to 1 decima. Sometimes full value is wanted. Especially when we use a 1Wire component to measure voltage and need that in HA. I personally use it in a template sensor to convert the value to a percentage gauge.

device_file attribute is perhaps not well choosen. device_id will be a better choise. The device_id would be of interest when sensor naming from config has been used. 

sensor_type is not needed.

Reply:
Can we instead use `device_file` as a unique_id? Does it match the requirements?

Then the user can rename entities and entity_id as wanted.

https://developers.home-assistant.io/docs/en/entity_registry_index.html#unique-id-requirements

Reply:
device_file uniquely identifies the sensor … that is the actual component and its input. That is probably the way to go.